design/oss_suite_release.md

```diff
@@ -0,0 +1,429 @@
+
+At current, we have no formal process for releasing the full suite of tools in
+the Conjur OSS ecosystem so that:
+- It is clear to end users which version of Conjur OSS they should use, and what
+  integrations and tooling are compatible with that version.
+- Key use cases that involve more than two components working together are automatically
+  tested to ensure that the end-to-end experience in those use cases works as expected.
+- Development can continue on the master branch of individual components while
+  end users have a reliable way to pull "stable" releases for all components and
+  avoid unexpectedly running "edge" releases .
+- It is clear what features are officially released and which are still in progress -
+  at current, there is no way to easily track unreleased features in our broad suite
+  of tools and integrations, and formally creating a system for tracking unreleased
+  features will help us to do a better job of comprehensively testing and releasing
+  completed features.
+
+In this effort, we will build a formal Conjur OSS Suite Release process to begin
+addressing these concerns.
+
+## Desired Outcome
+There is an initial Conjur OSS Suite Release that validates compatibility of the
+included components. In addition, the infrastructure and tooling exists for creating
+future OSS Suite Releases that may include additional automated tests and components.
+
+## Acceptance Criteria
+- All key components of the Conjur OSS Suite will be grouped into a single
+  Conjur OSS Suite Release.
+- There is a special [release repository](https://github.com/cyberark/conjur-oss-suite-release)
+  that is used to:
+  - Pin the versions of the components included in the suite release.
+  - Add test case for end-to-end flows involving components of the OSS suite.
+  - Automatically run test suite each day against component versions pinned in
+    the suite release.
+  - When the pinned component versions are changed and the suite version is
+    incremented:
+    - Automatically generate GitHub releases that include release notes with links
+      to artifacts.
+    - Automatically generate a documentation-friendly release page that includes
+      a table of contents, release notes for each component, and links to artifacts.
+  - Document how end users should utilize the Conjur OSS Suite Release.
+
+In addition, as part of building this release, we'll also be creating:
+- A [GitHub landing page](https://cyberark.github.io/conjur) that links to the
+  repos included in the Conjur OSS Suite and describes what each one does.
+- A template to be added to the READMEs of repos in the Conjur OSS Suite to clarify
+  about the state of the code on master ("edge") and to encourage end users to
+  visit the Conjur OSS Suite Release to determine which versions of each product
+  to use.
+- A tool for identifying the unreleased changes in the components of the Conjur OSS
```

Comment:
If possible I'd like to see a tool that covers the general diff case, allowing any two versions of the OSS suite release can be diffed. 

There are a few use cases for this:
* Generating release notes (showing only changes between the old and new releases specififed)
* Determining which changes an upgrade will bring, especially when skipping versions. 
* Determining unreleased changes (diff latest release against master)

Prior art: https://github.com/rcbops/osa_differ which generates a mark down report showing the differences between two openstack-ansible releases, including changes in all the sub projects.

Reply:
that sounds like a "nice to have", but based on the spike we're working on (cyberark/conjur-oss-suite-release#2) I think the team is considering this aspect!